cmd/pebble: nightly benchmarks failing

I manually ran the roachtest with `--debug-always` to be able to retrieve the contents of `init.log`:
```
wiping /mnt/data1/bench
dir /mnt/data1/bench
concurrency 1
panic: malformed key: 757365723130303030353236363730313237333432323000000000000000000109,

goroutine 98 gp=0xc040582380 m=8 mp=0xc040580008 [running]:
panic({0xc44a20?, 0xc00013ab60?})
        /usr/local/go/src/runtime/panic.go:779 +0x158 fp=0xc0515cb8d8 sp=0xc0515cb828 pc=0x43d198
github.com/cockroachdb/pebble/internal/crdbtest.Compare({0xc00013e2a0?, 0x0?, 0x0?}, {0x0, 0x0, 0x0})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/internal/crdbtest/crdbtest.go:173 +0x295 fp=0xc0515cb968 sp=0xc0515cb8d8 pc=0xa84af5
github.com/cockroachdb/pebble/internal/base.InternalCompare(0x0?, {{0xc00013e2a0?, 0x0?, 0x0?}, 0x0?}, {{0x0?, 0x0?, 0x0?}, 0x0?})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/internal/base/internal.go:369 +0x3d fp=0xc0515cb9b8 sp=0xc0515cb968 pc=0x76b59d
github.com/cockroachdb/pebble.(*pickedCompaction).maybeExpandBounds(0xc0003141e0, {{0xc00013e2a0?, 0xffff?, 0xd37d38?}, 0x0?}, {{0xc00013e420?, 0x0?, 0x0?}, 0x0?})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction_picker.go:384 +0x74 fp=0xc0515cba50 sp=0xc0515cb9b8 pc=0x9d9e54
github.com/cockroachdb/pebble.(*pickedCompaction).setupInputs(0xc0003141e0, 0x0?, 0xc0002ce908?, 0xc054754160)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction_picker.go:422 +0x39b fp=0xc0515cd2d0 sp=0xc0515cba50 pc=0x9da57b
github.com/cockroachdb/pebble.pickL0({0x5be9b85000, 0x5ea0b, 0xffffffffffffffff, {0x0, 0x0, 0x0}, {0xc0002e4288, 0xc0002e4290, 0x0}}, 0xc0002ce608, ...)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction_picker.go:1662 +0x1d5 fp=0xc0515cd9b8 sp=0xc0515cd2d0 pc=0x9e14f5
github.com/cockroachdb/pebble.(*compactionPickerByScore).pickAuto(0xc00035b380, {0x5be9b85000, 0x5ea0b, 0xffffffffffffffff, {0x0, 0x0, 0x0}, {0xc0002e4288, 0xc0002e4290, 0x0}})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction_picker.go:1255 +0x158 fp=0xc0515cfb88 sp=0xc0515cd9b8 pc=0x9defd8
github.com/cockroachdb/pebble.pickAuto({0xe153f8?, 0xc00035b380?}, {0x5be9b85000, 0x5ea0b, 0xffffffffffffffff, {0x0, 0x0, 0x0}, {0xc0002e4288, 0xc0002e4290, ...}})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1652 +0x5e fp=0xc0515cfbe8 sp=0xc0515cfb88 pc=0x9cd5fe
github.com/cockroachdb/pebble.(*DB).tryScheduleAutoCompaction(0xc0002e4008, {0x5be9b85000, 0x5ea0b, 0xffffffffffffffff, {0x0, 0x0, 0x0}, {0xc0002e4288, 0xc0002e4290, 0x0}}, ...)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1811 +0x34e fp=0xc0515cfd58 sp=0xc0515cfbe8 pc=0x9ceaee
github.com/cockroachdb/pebble.(*DB).maybeScheduleCompactionPicker(0xc0002e4008, 0xd373a8)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1743 +0x532 fp=0xc0515cfe68 sp=0xc0515cfd58 pc=0x9cdf12
github.com/cockroachdb/pebble.(*DB).maybeScheduleCompaction(...)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1648
github.com/cockroachdb/pebble.(*DB).flush.func1({0xe11b40?, 0xc00035c210?})
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1175 +0x1ba fp=0xc0515cff08 sp=0xc0515cfe68 pc=0x9ca37a
runtime/pprof.Do({0xe11a10?, 0x13dd3c0?}, {{0xc0001311a0?, 0x0?, 0xc00022e0c0?}}, 0xc000265fa8)
        /usr/local/go/src/runtime/pprof/runtime.go:51 +0x9d fp=0xc0515cff78 sp=0xc0515cff08 pc=0x7aea5d
github.com/cockroachdb/pebble.(*DB).flush(0xc000265fb8?)
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1153 +0x52 fp=0xc0515cffc8 sp=0xc0515cff78 pc=0x9ca192
github.com/cockroachdb/pebble.(*DB).maybeScheduleFlush.gowrap1()
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1069 +0x25 fp=0xc0515cffe0 sp=0xc0515cffc8 pc=0x9c9925
runtime.goexit({})
        /usr/local/go/src/runtime/asm_amd64.s:1695 +0x1 fp=0xc0515cffe8 sp=0xc0515cffe0 pc=0x474521
created by github.com/cockroachdb/pebble.(*DB).maybeScheduleFlush in goroutine 1
        /Users/jackson/go/src/github.com/cockroachdb/pebble/compaction.go:1069 +0xdb
```

We should update the roachtest to stick the init.log in the logs directory so it's automatically bundled within the artifacts.

Jira issue: PEBBLE-237

--- a/compaction_picker.go
+++ b/compaction_picker.go
@@ -374,23 +374,16 @@ func (pc *pickedCompaction) clone() *pickedCompaction {
 	return newPC
 }
 
-// maybeExpandedBounds is a helper function for setupInputs which ensures the
+// maybeExpandBounds is a helper function for setupInputs which ensures the
 // pickedCompaction's smallest and largest internal keys are updated iff
 // the candidate keys expand the key span. This avoids a bug for multi-level
 // compactions: during the second call to setupInputs, the picked compaction's
 // smallest and largest keys should not decrease the key span.
 func (pc *pickedCompaction) maybeExpandBounds(smallest InternalKey, largest InternalKey) {
-	emptyKey := InternalKey{}
-	if base.InternalCompare(pc.cmp, smallest, emptyKey) == 0 {
-		if base.InternalCompare(pc.cmp, largest, emptyKey) != 0 {
-			panic("either both candidate keys are empty or neither are empty")
-		}
+	if len(smallest.UserKey) == 0 && len(largest.UserKey) == 0 {
 		return
 	}
-	if base.InternalCompare(pc.cmp, pc.smallest, emptyKey) == 0 {
-		if base.InternalCompare(pc.cmp, pc.largest, emptyKey) != 0 {
-			panic("either both pc keys are empty or neither are empty")
-		}
+	if len(pc.smallest.UserKey) == 0 && len(pc.largest.UserKey) == 0 {
 		pc.smallest = smallest
 		pc.largest = largest
 		return

